mne/epochs.py

```diff
@@ -3249,6 +3260,38 @@ def __init__(
         # keep track of original sfreq (needed for annotations)
         raw_sfreq = raw.info["sfreq"]
 
+        # get events from annotations if no events given
+        if events is None:
+            events, event_id_tmp = events_from_annotations(raw)
+            if events.size == 0:
+                raise RuntimeError(
+                    "No usable annotations found in the raw object. "
+                    "Either `events` must be provided or the raw "
+                    "object must have annotations to construct epochs"
+                )
+            if any(raw.annotations.duration > 0):
+                logger.info(
+                    "Ignoring annotation durations and creating fixed-duration epochs "
+                    "around annotation onsets."
+                )
+            if event_id is None:
+                event_id = event_id_tmp
+            # if event_id is the names of events, map to events integers
+            if isinstance(event_id, str):
+                event_id = [event_id]
+            if isinstance(event_id, (list, tuple, set)):
+                if set(event_id).issubset(set(event_id_tmp)):
+                    event_id = {my_id: event_id_tmp[my_id] for my_id in event_id}
+                    # remove any non-selected annotations
+                    annotations.delete(
+                        ~np.isin(raw.annotations.description, list(event_id))
+                    )
+                else:
+                    raise RuntimeError(
+                        f"event_id(s) {set(event_id) - set(event_id_tmp)} "
+                        "not found in annotations"
+                    )
```

Comment:
The logic here should probably mimic what we do already for missing event IDs, e.g.,. having `Epochs(raw, event_id=["auditory", "visual"], on_missing="ignore")` should be okay when there are only `"auditory"` events in the annotations. Maybe you can remove some of the logic here to allow the existing `event_id+on_missing` checking/logic to take care of some stuff? (And please add a test for this case!)

Reply:
It looks like this comment was not addressed yet and was errantly resolved by GitHub because the code was moved

Reply:
Ahh I see there is an `_on_missing` in there now, never mind! Maybe we could deduplicate more at some point (?) but seems okay for now